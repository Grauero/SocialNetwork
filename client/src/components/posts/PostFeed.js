import React from 'react';
import propTypes from 'prop-types';

import PostItem from './PostItem';

const PostFeed = ({ posts }) => {
  return posts.map(post => <PostItem key={post._id} post={post} />);
};

PostFeed.propTypes = {
  posts: propTypes.instanceOf(Array).isRequired
};

export default PostFeed;
