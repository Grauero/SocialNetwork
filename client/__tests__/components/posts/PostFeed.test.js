import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import PostFeed from '../../../src/components/posts/PostFeed';

const props = {
  posts: [{ _id: '1', post: 'post1' }, { _id: '2', post: 'post2' }]
};
const component = shallow(<PostFeed {...props} />);

it('matches snapshot for 2 posts', () => {
  expect(toJSON(component)).toMatchSnapshot();
});
