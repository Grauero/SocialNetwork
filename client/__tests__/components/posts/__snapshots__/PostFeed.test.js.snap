// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for 2 posts 1`] = `
Array [
  <Connect(PostItem)
    key="1"
    post={
      Object {
        "_id": "1",
        "post": "post1",
      }
    }
  />,
  <Connect(PostItem)
    key="2"
    post={
      Object {
        "_id": "2",
        "post": "post2",
      }
    }
  />,
]
`;
